lib/pyinotify.py

```diff
@@ -0,0 +1,2346 @@
+        # Assuming we are watching at least for IN_CREATE and IN_DELETE
+        mask |= IN_CREATE | IN_DELETE
+
+        def cmp_name(event):
+            if getattr(event, 'name') is None:
+                return False
+            return basename == event.name
+        return self.add_watch(dirname, mask,
+                              proc_fun=proc_class(ChainIfTrue(func=cmp_name)),
+                              rec=False,
+                              auto_add=False, do_glob=False,
+                              exclude_filter=lambda path: False)
+
+    def get_ignore_events(self):
+        return self._ignore_events
+
+    def set_ignore_events(self, nval):
+        self._ignore_events = nval
+
+    ignore_events = property(get_ignore_events, set_ignore_events,
+                             "Make watch manager ignoring new events.")
+
+
+class RawOutputFormat:
+    """
+    Format string representations.
+    """
+    def __init__(self, format=None):
+        self.format = format or {}
+
+    def simple(self, s, attribute):
+        if not isinstance(s, str):
+            s = str(s)
+        return (self.format.get(attribute, '') + s +
+                self.format.get('normal', ''))
+
+    def punctuation(self, s):
+        """Punctuation color."""
+        return self.simple(s, 'normal')
+
+    def field_value(self, s):
+        """Field value color."""
+        return self.simple(s, 'purple')
+
+    def field_name(self, s):
+        """Field name color."""
+        return self.simple(s, 'blue')
+
+    def class_name(self, s):
+        """Class name color."""
+        return self.format.get('red', '') + self.simple(s, 'bold')
+
+output_format = RawOutputFormat()
+
+class ColoredOutputFormat(RawOutputFormat):
+    """
+    Format colored string representations.
+    """
+    def __init__(self):
+        f = {'normal': '\033[0m',
+             'black': '\033[30m',
+             'red': '\033[31m',
+             'green': '\033[32m',
+             'yellow': '\033[33m',
+             'blue': '\033[34m',
+             'purple': '\033[35m',
+             'cyan': '\033[36m',
+             'bold': '\033[1m',
+             'uline': '\033[4m',
+             'blink': '\033[5m',
+             'invert': '\033[7m'}
+        RawOutputFormat.__init__(self, f)
+
+
+def compatibility_mode():
+    """
+    Use this function to turn on the compatibility mode. The compatibility
+    mode is used to improve compatibility with Pyinotify 0.7.1 (or older)
+    programs. The compatibility mode provides additional variables 'is_dir',
+    'event_name', 'EventsCodes.IN_*' and 'EventsCodes.ALL_EVENTS' as
+    Pyinotify 0.7.1 provided. Do not call this function from new programs!!
+    Especially if there are developped for Pyinotify >= 0.8.x.
+    """
+    setattr(EventsCodes, 'ALL_EVENTS', ALL_EVENTS)
+    for evname in globals():
+        if evname.startswith('IN_'):
+            setattr(EventsCodes, evname, globals()[evname])
+    global COMPATIBILITY_MODE
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Global variables are dangerous because they can be simultaneously accessed from multiple sections of a program. This frequently results in bugs. Most bugs involving global variables arise from one function reading and acting on the value of a global variable before another function has the chance to set it to an appropriate value. We recommend using a configuration module to mutate global state.

[Learn more](https://docs.python.org/3/faq/programming.html#how-do-i-share-global-variables-across-modules)

Similar issue at line number 2291.